lib/src/main/java/com/basistheory/android/view/CardNumberElement.kt

```diff
@@ -21,14 +27,49 @@ class CardNumberElement : TextElement {
         attrs,
         defStyleAttr
     ) {
-      init()
+        init()
+    }
+
+    override fun afterTextChangedHandler(editable: Editable?) {
+        val cardResult = cardBrandEnricher.evaluateCard(getDigitsOnly(editable?.toString()))
+        if (cardResult.cardDetails?.cardMask != null)
+            mask = cardResult.cardDetails!!.cardMask.toList()
+
+        super.afterTextChangedHandler(editable)
     }
 
     private fun init() {
         super.keyboardType = KeyboardType.NUMBER
         super.mask = defaultMask
         super.transform = regexReplaceElementTransform(Regex("""\s"""), "")
         super.validate = ::luhnValidator
+
+        super.elementChangeEvent =
+            { value: String?, _: Boolean, isEmpty: Boolean, isValid: Boolean ->
+                val cardResult = cardBrandEnricher.evaluateCard(getDigitsOnly(value))
+                val eventDetails = cardResult.cardDetails?.brand?.let { b ->
+                    mutableListOf(
+                        EventDetails(
+                            "cardBrand",
+                            b
+                        )
+                    )
+                } ?: mutableListOf()
+
+                ChangeEvent(
+                    cardResult.complete ?: false,
+                    isEmpty,
+                    isValid,
+                    eventDetails
+                )
+            }
+
+        cardBrandEnricher = CardBrandEnricher()
```

Comment:
There's no reason not to init this where you declare it, is there? Then you can avoid making it a `lateinit var` and just make it a `val`